R/restore.R

```diff
@@ -223,302 +226,149 @@ getSourceForPkgRecord <- function(pkgRecord,
       }
     }
   } else if (identical(pkgRecord$source, "github")) {
-
-    # Prefer using https if possible. Note that 'wininet'
-    # can fail if attempting to download from an 'http'
-    # URL that redirects to an 'https' URL.
-    # https://github.com/rstudio/packrat/issues/269
-    method <- tryCatch(
-      secureDownloadMethod(),
-      error = function(e) "internal"
-    )
-
-    if (is.null(pkgRecord$remote_host) || !nzchar(pkgRecord$remote_host)) {
-      # Guard against packages installed with older versions of devtools
-      # (it's possible the associated package record will not contain a
-      # 'remote_host' entry)
-      protocol <- if (identical(method, "internal")) "http" else "https"
-      fmt <- "%s://api.github.com/repos/%s/%s/tarball/%s"
-      archiveUrl <- sprintf(fmt,
-                            protocol,
-                            pkgRecord$gh_username,
-                            pkgRecord$gh_repo,
-                            pkgRecord$gh_sha1)
-    } else {
-      # Prefer using the 'remote_host' entry as it allows for successfuly
-      # installation of packages available on private GitHub repositories
-      # (which will not use api.github.com)
-      fmt <- "%s/repos/%s/%s/tarball/%s"
-      archiveUrl <- sprintf(fmt,
-                            pkgRecord$remote_host,
-                            pkgRecord$remote_username,
-                            pkgRecord$remote_repo,
-                            pkgRecord$remote_sha)
-
-      # Ensure the protocol is prepended
-      if (!grepl("^http", archiveUrl)) {
-        protocol <- if (identical(method, "internal")) "http" else "https"
-        archiveUrl <- paste(protocol, archiveUrl, sep = "://")
-      }
-    }
+    archiveUrl <- githubArchiveUrl(pkgRecord)
 
     srczip <- tempfile(fileext = '.tar.gz')
-    on.exit({
-      if (file.exists(srczip))
-        unlink(srczip, recursive = TRUE)
-    }, add = TRUE)
+    on.exit(
+      if (file.exists(srczip)) unlink(srczip, recursive = TRUE),
```

Comment:
Is the existence check needed? Per help for `unlink`:

> Not deleting a non-existent file is not a failure, nor is being unable to delete a directory if recursive = FALSE.

Reply:
You're right. I just brought this over from the old code without questioning whether it was necessary. I'll remove it.